src/sinks/util/service.rs

```diff
@@ -163,7 +164,7 @@ pub struct TowerRequestLayer<L, Request> {
 impl<S, L, Request> tower::layer::Layer<S> for TowerRequestLayer<L, Request>
 where
     S: Service<Request> + Send + Clone + 'static,
-    S::Response: Send + 'static,
+    S::Response: Send + 'static + Response,
```

Comment:
The lifetime should go at the end, and probably this trait should go at the beginning before things like `Send`.